src/docfx/template/liquid/LiquidTemplate.cs

```diff
@@ -78,17 +66,37 @@ public string Render(string templateName, SourceInfo<string?> mime, JObject mode
                     formatProvider: CultureInfo.InvariantCulture),
             };
 
-            return template.Render(parameters);
+            try
+            {
+                t_package = _package;
+                return template.Render(parameters);
+            }
+            finally
+            {
+                t_package = default;
+            }
         }
 
-        public static string GetThemeRelativePath(DotLiquid.Context context, string resourcePath)
+        public static string? GetThemeRelativePath(string resourcePath)
         {
-            return Path.Combine((string)context["theme_rel"], resourcePath);
+            return t_package?.TryGetPhysicalPath(new PathString(resourcePath));
         }
 
-        private static Template LoadTemplate(string fullPath)
+        private Template LoadTemplate(PathString path)
         {
-            var template = Template.Parse(File.ReadAllText(fullPath));
+            return _templates.GetOrAdd(path, new Lazy<Template?>(() => LoadTemplateCore(path))).Value
+                ?? throw Errors.Template.LiquidNotFound(path).ToException(isError: false);
```

Comment:
>path [](start = 56, length = 4)

we have discussed this before, if we report a error message like `{layout}.liquid.js` not found, it may be confused to the end user since the layout is transport to the end user, so please keep the original error message based on `mime`